server: allow to use HDB backend

qbey opened the pull request:
Add an extra argument to the `LdapServer` class to allow use
of the HDB backend which among other things allow the use of
`SimplePagedResultsControl` as LDIF doesn't.

Commit:
server: allow to use HDB backend

Add an extra argument to the `LdapServer` class to allow use
of the HDB backend which among other things allow the use of
`SimplePagedResultsControl` as LDIF doesn't.

rbarrois commented:
I love this!

However, I'd rather go a step further: remove support for LDIF altogether — this is intended to be a simple yet faithful LDAP implementation ; I want to keep it as simple as possible.

What do you think?

Commit:
server: drop LDIF database backend support.

"It seems that perfection is attained, not when there is nothing
more to add, but when there is nothing more to take away."
Antoine de Saint Exupery
